ext/spl/spl_iterators.c

```diff
@@ -2038,7 +2038,7 @@ SPL_METHOD(RegexIterator, accept)
 
 	if (Z_TYPE(intern->current.data) == IS_UNDEF) {
 		RETURN_FALSE;
-	} else if (Z_TYPE(intern->current.data) == IS_ARRAY) {
+	} else if (intern->u.regex.flags ^ REGIT_USE_KEY && Z_TYPE(intern->current.data) == IS_ARRAY) {
```

Comment:
This should be `(intern->u.regex.flags & REGIT_USE_KEY)`. Alternatively the IS_ARRAY check can be moved into the branch in L2048.